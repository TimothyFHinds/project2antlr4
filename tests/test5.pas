program test5;

var  
num1 : REAL;
num2 : REAL;

//continue statement test cases
//https://www.tutorialspoint.com/pascal/pascal_continue_statement.htm
begin
    num1 := 10;

    while num1>0 do
        begin
            if(num1 = 5) then
                //skip iteration
                begin
                    num1 := num1 - 1;
                    continue;
                end;
            writeln('The value of num1: ');
            writeln(num1);
            writeln();  //write new line
            num1 := num1 - 1;
        end;

end.