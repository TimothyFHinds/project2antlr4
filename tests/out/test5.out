'While loop with continue, 3 should be missing'
10.0
9.0
8.0
7.0
6.0
5.0
4.0
2.0
1.0

'While loop with continue,  should print from 1 to 5'
1.0
2.0
3.0
4.0
5.0

'For loop with continue, output should go from 10 to 14'
10.0
11.0
12.0
13.0
14.0

'For loop with continue, output should skip 15'
10.0
11.0
12.0
13.0
14.0
16.0
17.0
18.0
19.0
20.0

